### PR TITLE
Blurry background workaround with canvas

### PR DESCRIPTION
**Pull request by trungk18**
https://user-images.githubusercontent.com/6767322/113589341-d1293700-9663-11eb-8889-8658ddc7cbcd.mp4

### CODE REVIEW

File: libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts [hunk 1]
```diff
@@ -1,23 +1,55 @@
...
-      if (!track?.album?.images) {
-        return null;
+export class AlbumArtOverlayComponent implements AfterViewInit {
+  @Input() set imageUrl(url: string) {
+    this.zone.runOutsideAngular(() => {
+      this.drawImageToCanvas(url);
+    });
+  }
+
+  @ViewChild('canvas') canvas!: ElementRef<HTMLCanvasElement>;
```
Comment (nartc) [inline libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts:26]: suggestion: `canvas` can be queried with `static: true` so it is available in `ngOnInit` instead.

File: libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts [hunk 2]
```diff
@@ -1,23 +1,55 @@
...
-      return imageUrl ? `url(${imageUrl})` : null;
-    })
-  );
+    };
+  }
 
-  constructor(private playbackStore: PlaybackStore) {}
+  initCanvas() {
+    this.context = this.canvas.nativeElement.getContext('2d');
+    if (this.context) {
```
Comment (nartc) [inline libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts:50]: nitpick: this if check looks like it's redundant.

File: libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts [hunk 3]
```diff
@@ -1,23 +1,55 @@
...
+  }
+
+  @ViewChild('canvas') canvas!: ElementRef<HTMLCanvasElement>;
+  context!: CanvasRenderingContext2D | null;
+
+  constructor(private zone: NgZone) {}
+
+  ngAfterViewInit() {
+    this.zone.runOutsideAngular(() => {
+      this.initCanvas();
```
Comment (nartc) [inline libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts:33]: suggestion: I would bringing the logic to initialize the canvas in `drawerImageToCanvas` as well. 

```ts
drawerImageToCanvas(imageUrl: string) {
   if (!this.context) {
      // initialize the context here if it's not available
   }
   // the rest of drawing Image to canvas logic
}
```

**Review by nartc — COMMENTED** [review]
Overall LGTM, just a couple of suggestions and nitpicks.

File: libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts (hunk 2 above)
Comment (trungk18) [reply libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts:50]: Noted, so we can just put `!` in `this.context!.`, right?

Comment (nartc) [reply libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts:50]: `context!: CanvasRenderingContext2D | null;` I didn't see this. this is redundant as well. if you plan to use `!` non-null assertion then might as well removing the `null` type from this.

File: libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts (hunk 3 above)
Comment (trungk18) [reply libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts:33]:
```ts
this.zone.runOutsideAngular(() => {
  if (!this.context) {
    this.initCanvas();
  }
  this.drawImageToCanvas(url);
})
```

**Pull request merged**